### ISSUE TITLE
When I am doing sequence labeling, I want to use the output label information at (t-1) step to be as a input of the next step (t). How can I do ?

### ISSUE THREAD

**Issue by TJUNLP** [keras-team/keras#9651]
When I am doing sequence labeling, Using LSTM, I want to use the output label information at (t-1) step to be as a input of the next step (t). How can I do ?
        
y(t) = LSTM(s(t-1), y(t-1), h(t))
    
Where s is the hidden state of the LSTM, y is the last output label information, and h is the hidden state at each time-steps from the encoder.

Maybe, I need to write a decode layer manually, but I do not know how to come true.
Please provide me something to learn, or could you tell me some methods else ?

**Comment by kris-singh** [comment]
I think you can use SimpleRNN for doing the above task from keras.
I looked at the source of SimpleRNN and SimpleRNNCell code but could not figure out how this is done since the step_function does a simple forward pass on the given input. Maybe some body can clear that out for me.

**Comment by tRosenflanz** [comment]
So as far as I am aware this is not possible in raw Keras because of how evaluation of operations is done (might be wrong about this). However, I have used [RecurrentShop ](https://github.com/farizrahman4u/recurrentshop) from @farizrahman4u to do this in the past. It is very easy to use!

**Comment by TJUNLP** [comment]
@tRosenflanz @farizrahman4u I could not use the RecurrentShop to solve my problem, could you show a demo ? thanks
